tests/rules/method.js

```diff
@@ -219,6 +220,32 @@ eslintTester.run("method", rule, {
             code: "(e = node.insertAdjacentHTML('beforebegin', '<s>safe</s>'))()",
             parserOptions: { ecmaVersion: 6 },
         },
+
+        // Typescript support tests
```

Comment:
Can you also supply tests that combine typescript with known-good usages of e.g., insertAdjacentHTML?

Reply:
done - Added some good test cases with known good usages for both rules!